enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -160,9 +160,13 @@
   "Returns the default index spec for a model."
   [embedding-model]
   (let [{:keys [model-name provider vector-dimensions]} embedding-model
-        ;; This is a hack to ensure we don't exceed postgres' limit of 63 chars for identifier names.
-        abbreviated-model-name (embedding/model->abbrev model-name model-name)
-        table-name (str "index_table_" provider "_" abbreviated-model-name "_" vector-dimensions)]
+        provider-name (str/replace provider #"[-:.]" "_")
+        table-name-prefix (str "index_table_" provider-name "_")
+        table-name-suffix (str "_" vector-dimensions)
+        ;; calculate remaining space to ensure we don't exceed postgres' limit of 63 chars for identifier names.
+        model-max-name-len (- 63 (count table-name-prefix) (count table-name-suffix))
```

Comment:
This only prevents the table-name from exceeding 63 chars, but doesn't help with index names, which have the table-name as a prefix, right? So if the table-name winds up using all 63 chars, all of the index names will collide?

Also fwiw, postgres will silently truncate the names if they exceed the limit, so not sure we need to truncate ourselves (though maybe it at least makes it clear it's happening). The idea behind the abbreviations was to make sure we got unique, non-truncated names for all the indexes.